src/core/GameKeyHandler.ttslua

```diff
@@ -135,11 +135,17 @@ function discardObject(playerColor, hoveredObject)
     return
   end
 
-  -- warning for locations since these are usually not meant to be discarded
-  if hoveredObject.hasTag("Location") then
-    broadcastToAll("Watch out: A location was discarded.", "Yellow")
+  -- These should probably not be discarded normally. Ask player for confirmation.
+  if (hoveredObject.type == "Deck") or hoveredObject.hasTag("Location") then
+    local suspect = (hoveredObject.type == "Deck") and "Deck" or "Location"
+    Player[playerColor].showConfirmDialog("Discard " .. suspect .. "?", function () performDiscard(hoveredObject, playerColor) end)
+    return
   end
 
+  performDiscard(hoveredObject, playerColor)
+end
+
+function performDiscard(hoveredObject, playerColor)
```

Comment:
I'd like to keep the order of the arguments the same for the functions (playerColor -> hoveredObject). Looks good otherwise 👍